src/code.sh

```
@@ -4,149 +4,260 @@
+    MEM_IN_MB="$(("${MEM}"/1024))m"
+
+    tar zxvf "$fasta_index_path"
+    OUTPUT_DIR="${HOME}/out/eggd_picard_stats/QC"
+    mkdir -p "$OUTPUT_DIR"
+
+    # Create the interval file if required
+    if [[ "$run_CollectMultipleMetrics" == true ]] || \
+        [[ "$run_CollectHsMetrics" == true ]] || \
+        [[ "$run_CollectTargetedPcrMetrics" == true ]]; then
+        echo "Generating interval file"
+        create_interval_file "${bedfile_path}" "${sorted_bam_path}" targets.picard "${MEM_IN_MB}"
```

Shellcheck (0.10.0)

[warning] bedfile_path is referenced but not assigned (did you mean 'BEDFILE_PATH'?).

(SC2154)